doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md

```diff
@@ -0,0 +1,168 @@
+**Description**: AP staff member or area manager  
+**Authorised to**:
+- record arrival
+- record non-arrival
+- extend a booking
+- view or cancel a booking
+- view / edit / create lost beds
+- view capacity (occupancy)
+- record departure
+- view residents
+
+#### Admin
+**Role**: `ROLE_ADMIN`  
+**Description**: Manages AP role  
+**Authorised to**:
+- add AP roles to Delius users
+- remove AP roles from Delius users
+
+### The "applicant role"
+
+An "applicant" needs access to more than their individual caseload, in order to create and
+access applications for people managed by fellow team members.
+
+We know that probation officers should have membership of only one team, but currently
+this is not the case. At present, approx 270 staff in the North East region are members of multiple teams.
+
+When restricting our "Applicant" role to CRNs which are within the caseload of their
+team(s) we are considering the following scenarios:
+
+1. applicant creates a new application for a given CRN 
+2. applicant views existing applications, created by themselves 
+3. applicant views existing applications, created by team members
+
+
+## Decision
+
+We're basing our applicant access control on the caseload(s) of the team(s) in which the
+applicant has membership.
+
+### 1. Creating an application
+
+To achieve [1] we're asking the probation-integration (PI) team to give us a new endpoint on the
+[approved-premises-and-delius](https://github.com/ministryofjustice/hmpps-probation-integration-services/tree/main/projects/approved-premises-and-delius) API which they've built.
+
+For example, something like:
+
+```sh
+curl "https://approved-premises-and-delius/{crn}/managingTeams/{delius-staffcode}"
+```
+
+could return a list of teams in which i) the given Delius staff member has membership and 
+ii) which has the given CRN in its caseload. For example a list of "managing teams":
+
+```json
+{  
+	"managingTeamCodes": ["N01ABC"]
+}
+```
+
+or alternatively perhaps a list of the staff member teams with a boolean `isOffenderInCaseload` 
+property to indicate whether the given CRN is managed by that team:
+
+```json
+[
+  {
+    "teamCode": "N02CBA", 
+    "isOffenderInCaseload": false
+  },
+  {
+    "teamCode": "N01ABC", 
+    "isOffenderInCaseload": true
+  }
+]
+```
+
+
+### 2. Listing applications created by team
+
+Search for applications with a team ID matching the membership of the logged-in user 
+(`applications.created_by_team_id` ?).
+
+NB: the service needs to start storing "team code" as a property on the `applications` table.
+
+### 3. Listing applications created by user
+
+Search for applications created by the logged-in user (`applications.created_by_user_id`). 
+Actually viewing the application could be restricted to people who are currently a member 
+of a team managing the CRN.
+
+## Consequences
+
+There are a couple of open questions:
+
+1. what do we do in the case of applicants who are members of more than one team? Do we:
```

Comment:
I don't think this is as much of an issue anymore as we will know which of the user's teams manages the Offender from the response from `https://approved-premises-and-delius/{crn}/managingTeams/{delius-staffcode}`.

The edge case is reduced to: the user is a member of more than one team where more than one of those teams has the Offender in its caseload.  Ideally I think we would ask the user to select the relevant team but given time constraints perhaps just storing all of the teams is the way to go here?

Reply:
I think we'll need to understand if there is ever a case where an offender is in more than one caseload first? It seems to me like it'd be a 1:1 relationship right?

Reply:
I think this depends on the answer to point 2.  i.e. an Offender can appear in the caseload for multiple teams at more granular levels so it depends which level of granulairty the integrations team endpoint operates at.  If I recall correctly this was examplified as part of their probation terms might be the responsibilit of a different team to the one that bares overall responsibility for them

Reply:
Yes I believe that more than one team can have a person in their caseload so the endpoint may return more than one team. And it sounds like storing a list of team codes against each application (e.g. `application.managing_team_codes`) would be a good solution.

Reply:
I've added this to the draft